fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordMetaDataBuilder.java

```diff
@@ -966,6 +1071,36 @@ public void setVersion(int version) {
         this.version = version;
     }
 
+    /**
+     * Enable counter-based subspace key mechanism. If enabled, index subspace
+     * keys will be set based on a counter instead of index name.
+     *
+     * <p>
+     * Existing clients should be careful about using counter-based subspace keys. The name of an index is the default
+     * value of its subspace key when counter-based subspace keys are disabled. If a subspace key was not set
+     * explicitly before, enabling the counter-based scheme will change the index's subspace key to a counter. Note that
+     * it is important that the subspace key of an index that has data, does not change.
+     * See {@link Index#setSubspaceKey(Object)} for more details.
+     * </p>
+     *
+     * @return this builder
+     */
+    public RecordMetaDataBuilder enableCounterBasedSubspaceKey() {
+        if (recordsDescriptor != null) {
+            throw new MetaDataException("Records descriptor has already been set.");
+        }
+        this.useCounterBasedSubspaceKey = true;
+        return this;
+    }
+
+    /**
+     * Checks if counter-based subspace key mechanism is enabled.
+     * @return true if counter-based subspace key is enabled
+     */
+    public boolean counterBasedSubspaceKeyEnabled() {
```

Comment:
This method and `RecordMetaData::isCounterBasedSubspaceKeyEnabled` should have the same name. I'm not sure what the name should be, necessarily.

Reply:
used isCounterBasedSubspaceKeyEnabled.